### PR TITLE
[ticket/15413] Login redirect to previous page

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-15413

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15413

### CODE REVIEW

File: phpBB/includes/functions.php [hunk 1]
```diff
@@ -4181,7 +4181,7 @@ function page_header($page_title = '', $display_online_list = false, $item_id =
 	}
 	else
 	{
-		$u_login_logout = append_sid("{$phpbb_root_path}ucp.$phpEx", 'mode=login');
+		$u_login_logout = append_sid("{$phpbb_root_path}ucp.$phpEx", 'mode=login&amp;redirect=' . rawurlencode($user->page['page']));
```
Comment (DavidIQ) [inline phpBB/includes/functions.php]: What happens if I enter my password wrong and then enter it correctly?

Comment (rubencm) [reply phpBB/includes/functions.php]: Done (i think)

**Comment by rubencm** [comment]
!set WIP

**Comment by OlaRossum** [comment]
I got moved to pulls, when I logged in

**Comment by DavidIQ** [comment]
What is "pulls"? @OlaRossum

**Comment by OlaRossum** [comment]
It's this site I got redirected to: https://github.com/phpbb/phpbb/pulls
From: https://github.com/phpbb/phpbb

**Comment by DavidIQ** [comment]
Absolutely nothing to do with this pull request.

**Comment by OlaRossum** [comment]
Ok. You know how I can get this to my master? Or do I have to upload all the files? I haven't even leaned to upload a directory.

**Comment by rubencm** [comment]
Don't use, it still doesn't work, and if it works, you can't login to the acp or something like that, i don't remember

File: phpBB/includes/functions.php (hunk 1 above)
Comment (OlaRossum) [reply phpBB/includes/functions.php]: landed on https://github.com/phpbb/phpbb when I was logged into Github

Comment (DavidIQ) [reply phpBB/includes/functions.php]: Again nothing to do with this pull request. This pr is for the phpBB software and wouldn't redirect you to GitHub.

Comment (OlaRossum) [reply phpBB/includes/functions.php]: Yeah, I figured that out when I saw where the page landed..
I really like the emotion buttons. That's why I want to learn more about this forum.

File: phpBB/includes/functions.php [hunk 2]
```diff
@@ -2324,7 +2324,8 @@ function login_box($redirect = '', $l_explain = '', $l_success = '', $admin = fa
 		// The result parameter is always an array, holding the relevant information...
 		if ($result['status'] == LOGIN_SUCCESS)
 		{
-			$redirect = $request->variable('redirect', "{$phpbb_root_path}index.$phpEx");
+			$redirect = $request->variable('redirect', "index.$phpEx", false, \phpbb\request\request_interface::POST);
```
Comment (gooof) [inline phpBB/includes/functions.php]:
```php
$redirect = $admin ? "{$phpbb_root_path}index.$phpEx" : $request->variable('redirect', "index.$phpEx", false, \phpbb\request\request_interface::POST);

File: phpBB/includes/functions.php (hunk 1 above)
Comment (gooof) [reply phpBB/includes/functions.php]:
```php
// if user enter password wrong and then correct
$redirect = !empty($request->variable('redirect', '')) ? $request->variable('redirect', '') : $user->page['page'];
$u_login_logout = append_sid("{$phpbb_root_path}ucp.$phpEx", 'mode=login&amp;redirect=' . rawurlencode($redirect));

**Comment by rubencm** [comment]
!unset WIP

**Comment by GerB** [comment]
```
empty($request->variable('redirect', ''))
```
This wil result in a compile error on PHP 5.x

I've proposed an improvement to your code here: https://github.com/phpbb/phpbb/pull/5140
Good work though.

**Comment by rubencm** [comment]
Ok, thanks, i close this

**Comment by rubencm** [comment]
@GerB Anyway, this goes for phpBB3.3, whose minimal version of php will be 7.0 i think

**Comment by GerB** [comment]
@rubencm 5.6 is the target for 3.3 AFAIK: https://area51.phpbb.com/phpBB/viewtopic.php?f=81&t=50831&start=20#p301751

**Comment by rubencm** [comment]
Ah ok, didn't know that

**Pull request closed without merging**